### PR TITLE
fix(redux-actions): createAction<Payload>(actionType: string) is incorrect.

### PR DESCRIPTION
**Pull request by meowtec**
createAction<Payload>(actionType: string) and createAction(actionType: string) is incorrect

also see #15697

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

File: types/redux-actions/index.d.ts [hunk 1]
```diff
@@ -85,7 +89,7 @@ export function createAction<Payload, Arg1, Arg2, Arg3, Arg4>(
 
 export function createAction<Payload>(
     actionType: string
-): ActionFunctionAny<Action<Payload>>;
+): ActionFunction1<Payload, Action<Payload>>;
```
Comment (andy-ms) [inline types/redux-actions/index.d.ts]: This can't be right -- now there are 2 single-argument and they return different results.

Comment (meowtec) [inline types/redux-actions/index.d.ts]: I find a similar example:
```
  public static create<T0>(): Dispatcher1<T0>;
  public static create<T0, T1>(): Dispatcher2<T0, T1>;
  public static create<T0, T1, T2>(): Dispatcher3<T0, T1, T2>;
```
from https://github.com/Microsoft/TypeScript/issues/1024#issuecomment-68118987 .

And I also test my new definition:

```
const act0 = ReduxActions.createAction('ACTION0')
const act1 = ReduxActions.createAction<string>('ACTION1')

act0().payload // void
act1('hello').payload // string
```

It seem to be expected.

**Comment by andy-ms** [comment]
Thanks!

**Comment by meowtec** [comment]
I think it will be better like this:

```
export function createAction(
    actionType: string
): ActionFunctionAny<Action<void>>; // ActionFunctionAny instead of ActionFunction0
```

Now this code will get an error:
```
createAction('ACTION')('payload')
```

**Comment by andy-ms** [comment]
Sound good, you can make a PR.

**Pull request merged**